doc/modules.md

```diff
@@ -0,0 +1,96 @@
+Modularization
+============
+
+This document describes the changes for a more fine grained modularization
+of common libraries, e.g., Lunchbox
+
+## Requirements
+
+* Per-functionality targets
+* Improved build times for super projects
+
+## Implementation
+
+* All sub project targets are excluded from the default build, and
+  required targets are built only based on the dependencies.
```

Comment:
What if you are building Lunchbox as a standalone project to install the libraries in your system and use them from there afterwards?